### DISCUSSION TITLE
Is there a way to simulate a simple SIR model to Iris

### DISCUSSION THREAD

**Discussion by ripatti** [IRIS-Solutions-Team/IRIS-Toolbox#332] [General]
I wonder if there is a way to simulate a simple SIR model with Iris? 

The model code would like like this
```
%% Definitions
!transition-variables 
  S, I, R

!parameters 
  pi_r = 1/14; % per-period recovery rate
  pi = 2.5/14; % =R0*pi_r


%% Model block
!transition_equations
  S =# S{-1} - pi*I{-1}*S{-1};
  I =# I{-1} + pi*I{-1}*S{-1} - pi_r*I{-1} ;
  R =# R{-1} + pi_r*I{-1};
```
And the code to run
```
m = Model.fromFile('SIR.model','linear', false,'comment','SIR model');
len = 150;
idb = struct();
idb.I = 0.001;
idb.S = 1-idb.I;
idb.R = 1 - idb.S - idb.I;
R0 = m.pi/m.pi_r;
db = simulate(m, idb,1:len,'method','period');
```
But the last option does not work because we have not solved the model. There is no steady state either.

**Comment by jaromir-benes** [comment]
Hi - the first order solution is used by default to start iterations when
method=stacked or method=period. Use startIterations="data" and supply some
initial values (time series on the simul range) for all variables in the
input databank.

Best
J

**Reply by ripatti** [reply]
Thanks a lot!

Although these SIR models are easy to solve and can be programmed easily, Iris provides the wide set of other tools (consider, eg, different parameterizations) to experiment with the model. Therefore, I try to use Iris in performing any model simulation or development. 

This modification works:
```
idb.I = Series(1:len,0.001.*ones(1,len));
.
.
.
db = simulate(m, idb,2:len,'method','period', 'startIterationsfrom','data');

```
the range has to start from 2 to get the initial state information.

**Reply by ripatti** [reply]
It seems that assigning multiple parameter values does not work with this model object? 
eg
```
p.pi = [0.1 0.2];
assign(m,p); 
```
does not work. (latest version of Iris)

**Comment by jaromir-benes** [comment]
m = alter(m, 2);
m.pi = [ ...];
